modules/storages/lib/api/v3/file_links/create_endpoint.rb

```diff
@@ -0,0 +1,44 @@
+module API::V3::FileLinks
+  class CreateEndpoint < API::Utilities::Endpoints::Create
+    include ::API::V3::Utilities::Endpoints::V3Deductions
+    include ::API::V3::Utilities::Endpoints::V3PresentSingle
+
+    def process(request, params_elements)
+      global_result = ServiceResult.new(
+        success: true,
+        result: []
+      )
+      params_elements.each do |params|
+        one_result = super(request, params)
+        global_result.add_dependent!(one_result)
+        global_result.result << one_result.result
+      end
+      global_result
+    end
+
+    def present_success(request, call)
+      render_representer.create(
+        call.result,
+        self_link: request.api_v3_paths.file_links(request.work_package.id),
```

Comment:
The self link denotes the location where the resource is to be found. So a client can expect to have the same set of resources returned if it requests the self-url (not true if other resources have been added to the collection, tbh). This is not true here, since the POST will in fact work as a PATCH when considering that the created `FileLinks` will be added to the set of already existing ones instead of replacing them. 

We haven't had that before, but can we remove the self link in this case? That might also be weird since we always return a `self` link. 

To have that solved correctly, we would need to add a filtering to the self link with the ids of the newly created `FileLinks` as filter values. 

This would only be for the sake of being 'correct', though, so it might be better to be pragmatic and not 100% correct. Probably depends on the effort.

Reply:
yes, we had the same remarks (can't remember if it was with Eric or Oliver), and we took the pragmatic approach of having a self link without any filters, and modify it later if needed/requested during review.

So which approach should we prefer?

Reply:
I think it is ok to leave it as it is. I don't expect any client to depend on it.